File: Mobile/src/com/esprit/app/entity/Article.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.esprit.app.entity;

public class Article {
    private int id;
    private String contenu;
    private String titre;
    private String etat;
    private String image;
    private int nbrreact;
    private int auteur;

    public Article() {
    }

    public Article(String contenu, String titre, String etat, String image, int nbrreact, int auteur) {
        this.contenu = contenu;
        this.titre = titre;
        this.etat = etat;
        this.image = image;
        this.nbrreact = nbrreact;
        this.auteur = auteur;
    }

    public Article(int id, String contenu, String titre, String etat, String image, int nbrreact, int auteur) {
        this.id = id;
        this.contenu = contenu;
        this.titre = titre;
        this.etat = etat;
        this.image = image;
        this.nbrreact = nbrreact;
        this.auteur = auteur;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getContenu() {
        return contenu;
    }

    public void setContenu(String contenu) {
        this.contenu = contenu;
    }

    public String getTitre() {
        return titre;
    }

    public void setTitre(String titre) {
        this.titre = titre;
    }

    public String getEtat() {
        return etat;
    }

    public void setEtat(String etat) {
        this.etat = etat;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public int getNbrreact() {
        return nbrreact;
    }

    public void setNbrreact(int nbrreact) {
        this.nbrreact = nbrreact;
    }

    public int getAuteur() {
        return auteur;
    }

    public void setAuteur(int auteur) {
        this.auteur = auteur;
    }

    @Override
    public String toString() {
        return "Article{" + "contenu=" + contenu + ", titre=" + titre + ", etat=" + etat + ", image=" + image + ", nbrreact=" + nbrreact + ", auteur=" + auteur + '}';
    }
    
    
}

File: Web/src/Entity/Article.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

use Symfony\Component\Validator\Constraints as Assert;

/**
 * @ORM\Entity(repositoryClass="ArticleRepository::class")
 */
class Article
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(name="idArticle", type="integer")
     */
    private $idarticle;

    /**
     * @var string
     * @ORM\Column(name="contenu", type="text", length=200, nullable=false)
     * @Assert\NotBlank(message="description  doit etre non vide")
     * @Assert\Length(
     *      min = 7,
     *      max = 100,
     *      minMessage = "doit etre >=7 ",
     *      maxMessage = "doit etre <=100" )
     */
    private $contenu;

    /**
     * @var string
     * @ORM\Column(name="titre", type="string", length=15, nullable=false)
     * @Assert\NotBlank(message=" titre doit etre non vide")
     * @Assert\Length(
     *      min = 5,
     *      minMessage=" Entrer un titre au mini de 5 caracteres"
     *
     *     )
     */
    private $titre;

    /**
     * @var string
     *
     * @ORM\Column(name="Etat", type="string", length=200, nullable=false)
     */
    private $etat;

    /**
     * @var string
     *
     * @ORM\Column(name="image", type="string", length=200, nullable=false)
     */
    private $image;

    /**
     * @var int
     *
     * @ORM\Column(name="nbrreact", type="integer", nullable=false)
     */
    private $nbrreact;

    /**
     * @var \User
     *
     * @ORM\ManyToOne(targetEntity="User")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="auteur", referencedColumnName="idUser")
     * })
     */
    private $auteur;

    public function getIdarticle(): ?int
    {
        return $this->idarticle;
    }

    public function getContenu(): ?string
    {
        return $this->contenu;
    }

    public function setContenu(string $contenu): self
    {
        $this->contenu = $contenu;

        return $this;
    }

    public function getTitre(): ?string
    {
        return $this->titre;
    }

    public function setTitre(string $titre): self
    {
        $this->titre = $titre;

        return $this;
    }

    public function getEtat(): ?string
    {
        return $this->etat;
    }

    public function setEtat(string $etat): self
    {
        $this->etat = $etat;

        return $this;
    }

    public function getImage()
    {
        return $this->image;
    }

    public function setImage($image)
    {
        $this->image = $image;

        return $this;
    }

    public function getNbrreact(): ?int
    {
        return $this->nbrreact;
    }

    public function setNbrreact(int $nbrreact): self
    {
        $this->nbrreact = $nbrreact;

        return $this;
    }

    public function getAuteur(): ?User
    {
        return $this->auteur;
    }

    public function setAuteur(?User $auteur): self
    {
        $this->auteur = $auteur;

        return $this;
    }


}
